docs/rule/simple-modifiers.md

```diff
@@ -0,0 +1,34 @@
+# simple-modifiers
+
+This rule strongly advises against using the `{{modifier}}` helper with its first argument being either:
+
+- a string literal (.e.g. `{{modifier "tracking-interaction"}}`)
+- a variable in the template's JS backing class context (e.g. `{{modifier this.trackingInteraction}}`)
+
+A common misuse of this rule is to apply the modifier conditionally `{{(modifier (unless this.hasBeenClicked "track-interaction") "click" customizeData=this.customizeClickData)}}`, which technically is not wrong, since the helper `modifier` simply ignores `null` and `undefined`, producing a no-op modifier.
```

Comment:
```suggestion
A common issue this rule will catch is declaring the modifier's name conditionally like `{{(modifier (unless this.hasBeenClicked "track-interaction") "click" customizeData=this.customizeClickData)}}`. This is technically correct since the helper `modifier` simply ignores `null` and `undefined` values producing a no-op modifier but it is confusing. Ideally the we should instead do `{{(unless this.hasBeenClicked (modifier "track-interaction" "click" customizeData=this.customizeClickData))}}` where it's clear the modifier will not get called.
```

Trying to make it easier to understand the motivation here ^